openedx/core/djangoapps/user_authn/views/login_form.py

```diff
@@ -71,31 +68,25 @@ def login_and_registration_form(request, initial_mode="login"):
     if '?' in redirect_to:
         try:
             next_args = six.moves.urllib.parse.parse_qs(six.moves.urllib.parse.urlparse(redirect_to).query)
-            provider_id = next_args['tpa_hint'][0]
-            tpa_hint_provider = third_party_auth.provider.Registry.get(provider_id=provider_id)
-            if tpa_hint_provider:
-                if tpa_hint_provider.skip_hinted_login_dialog:
-                    # Forward the user directly to the provider's login URL when the provider is configured
-                    # to skip the dialog.
-                    if initial_mode == "register":
-                        auth_entry = pipeline.AUTH_ENTRY_REGISTER
-                    else:
-                        auth_entry = pipeline.AUTH_ENTRY_LOGIN
-                    return redirect(
-                        pipeline.get_login_url(provider_id, auth_entry, redirect_url=redirect_to)
-                    )
-                third_party_auth_hint = provider_id
-                initial_mode = "hinted_login"
+            if 'tpa_hint' in next_args:
```

Comment:
**Note to reviewers:** I added this extra `if` statement here since it is an expected condition for `tpa_hint` to not be included as a parameter.  We receive on the order 100k exceptions a day due to this lack of check.